core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -351,6 +352,7 @@ class TransactionCoordinator(brokerId: Int,
                            producerId: Long,
                            producerEpoch: Short,
                            txnMarkerResult: TransactionResult,
+                           fromClient: Boolean,
```

Comment:
I think reusing `handleEndTransaction` to handle coordinator aborts has always been a little confusing. As an alternative, we could move the logic here to an internal helper, maybe just named `endTransaction`. Then we can assume calls from `handleEndTransaction` are from the client.